api/src/test/java/org/openmrs/module/fhir2/api/search/ObservationSearchQueryTest.java

```diff
@@ -1585,6 +1585,112 @@ public void searchForLastnObs_shouldHandleNoCodeRequest() {
 		assertThat(resultList, everyItem(anyOf(allOf(is(instanceOf(Observation.class))))));
 	}
 	
+	@Test
+	public void searchForLastnEncountersObs_shouldHandleNormalRequest() {
+		ReferenceAndListParam referenceParam = new ReferenceAndListParam();
+		ReferenceParam patient = new ReferenceParam();
+		
+		patient.setValue(PATIENT_UUID);
+		
+		referenceParam.addValue(new ReferenceOrListParam().add(patient));
+		
+		TokenAndListParam categories = new TokenAndListParam().addAnd(new TokenParam().setValue("laboratory"));
+		
+		TokenAndListParam code = new TokenAndListParam().addAnd(
+		    new TokenParam().setSystem(FhirTestConstants.LOINC_SYSTEM_URL).setValue(LOINC_SYSTOLIC_BP),
+		    new TokenParam().setSystem(FhirTestConstants.CIEL_SYSTEM_URN).setValue(OBS_CONCEPT_ID));
+		
+		SearchParameterMap theParams = new SearchParameterMap().addParameter(FhirConstants.CODED_SEARCH_HANDLER, code)
+		        .addParameter(FhirConstants.CATEGORY_SEARCH_HANDLER, categories)
+		        .addParameter(FhirConstants.MAX_SEARCH_HANDLER, new NumberParam(2))
```

Comment:
One test can be added for max = 1 to test the difference in outputs.

Reply:
Yes, I'll add it